src/components/dex/Pair.vue

```diff
@@ -20,50 +20,104 @@
+import BaseSelector from './BaseSelector';
+
+const TABLE_COLUMNS = ['asset', 'price', 'volume', '24H change'];
+let storeUnwatch;
 
 export default {
-  mounted() {
-    this.loadTrades();
+  beforeDestroy() {
+    storeUnwatch();
   },
 
   components: {
     MarketPairChart,
-    MarketSelector,
+    BaseSelector,
   },
 
   computed: {
-    //
+    baseCurrencies() {
+      return _.uniq(_.map(this.$store.state.markets, 'baseCurrency'));
+    },
+
+    percentChangeAbsolute() {
+      return this.$store.state.tradeHistory ?
+        Math.abs(this.$store.state.tradeHistory.change24HourPercent) : 0;
+    },
+
+    tableData() {
+      return this.filteredMarkets().map(({ quoteCurrency }) => {
+        // TODO: This needs to be improved with real data.
+        const tradeHistory = this.$store.state.tradeHistory;
+        const price = this.$formatTokenAmount(tradeHistory ? tradeHistory.close24Hour : 0);
+        const vol = this.$formatNumber(tradeHistory ? tradeHistory.volume24Hour : 0);
+        const change = this.$formatNumber(this.percentChangeAbsolute);
+        return { asset: quoteCurrency, price, volume: vol, '24H change': change };
+      });
+    },
   },
 
   data() {
     return {
+      baseCurrency: '',
+      columns: TABLE_COLUMNS,
       searchBy: '',
-      data: TICKER_LIST.DATA,
-      columns: TICKER_LIST.COLUMNS,
     };
   },
 
   methods: {
-    loadTrades() {
-      this.$store.dispatch('fetchTradeHistory', {
-        marketName: this.$store.state.currentMarket.marketName,
+    filteredMarkets() {
+      return this.$store.state.markets.filter((market) => {
+        if (this.searchBy.length > 0) {
+          const isSearchMatch = _.includes(market.quoteCurrency, this.searchBy.toUpperCase());
+          return market.baseCurrency === this.baseCurrency && isSearchMatch;
+        }
+        return market.baseCurrency === this.baseCurrency;
       });
     },
+
     formatEntry(value, entry, key) {
       if (key === '24H change') {
         return `${this.$services.formatting.formatNumber(value)}%`;
       }
       return value;
     },
+
     getRelativeChange(value, entry, key) {
       if (key === '24H change') {
         return value > 0 ? 'increase' : 'decrease';
       }
       return null;
     },
+
+    handleMarketSelection({ asset }) {
+      this.$store.commit('setCurrentMarket', this.filteredMarkets().find((market) => {
+        return market.quoteCurrency === asset;
+      }));
+    },
+
+    loadTrades() {
+      if (!this.$store.state.currentMarket) {
+        return;
+      }
+
+      this.$store.dispatch('fetchTradeHistory', {
+        marketName: this.$store.state.currentMarket.marketName,
+      });
+    },
+  },
+
+  mounted() {
+    this.loadTrades();
+    this.baseCurrency = _.first(this.baseCurrencies);
+
+    storeUnwatch = this.$store.watch(
+      () => {
+        return this.$store.state.currentMarket;
+      }, () => {
+        this.loadTrades();
+      });
```

Comment:
@crishellco, This is what I was asking about this last weekend. I'm not sure what this is doing. It seems like it's setting up a session of 'watching' a store value. Why can't this be done in the watch methods?

Reply:
Yeah, so this _should_ work with a normal `watch`

```
...mapGetters(['currentMarket']),

Reply:
I don't understand what is going on here. I'm not sure these equal each other. I'm leaving it as is, but feel free to mess with it in another PR.